odoo/models.py

```diff
@@ -3276,14 +3276,13 @@ def write(self, vals):
                             self._name, ', '.join(sorted(unknown_names)))
 
         with self.env.protecting(protected_fields, self):
-            # write stored fields with (low-level) method _write
-            if store_vals or inverse_vals or inherited_vals:
-                # if log_access is enabled, this updates 'write_date' and
-                # 'write_uid' and check access rules, even when old_vals is
-                # empty
-                self._write(store_vals)
+            # update references to parents
+            ref_store_vals = {k: v for k, v in store_vals.items()
+                              if k in self._inherits.values()}
```

Comment:
`self._inherits` is pretty small in all cases, even empty in most cases.
I would therefore iterate on `self._inherits` instead of `storevals`:
```python
ref_store_vals = {k: store_vals[k] for k in self._inherits.values() if k in store_vals}
```

Reply:
that's an honky good idea !